### PR TITLE
TabPanel: Add tabsPosition option

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

File: packages/devextreme/js/ui/tab_panel.js [hunk 1]
```diff
@@ -252,6 +273,51 @@ const TabPanel = MultiView.inherit({
...
     },
 
+    _getTabsOrientation() {
+        const { tabsPosition } = this.option();
+
+        if([TABS_POSITION.top, TABS_POSITION.bottom].includes(tabsPosition)) {
+            return TABS_ORIENTATION.horizontal;
+        }
+
+        if([TABS_POSITION.right, TABS_POSITION.left].includes(tabsPosition)) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/tab_panel.js:279]: I think we may simplify this and return `TABS_ORIENTATION.vertical` immediately

File: packages/devextreme/js/ui/tab_panel.js [hunk 2]
```diff
@@ -252,6 +273,51 @@ const TabPanel = MultiView.inherit({
         this._focusTarget().attr('tabIndex', -1);
     },
 
+    _getTabsOrientation() {
+        const { tabsPosition } = this.option();
+
+        if([TABS_POSITION.top, TABS_POSITION.bottom].includes(tabsPosition)) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/tab_panel.js]:
```
if(tabsPosition === TABS_POSITION.top || tabsPosition === TABS_POSITION.bottom) {
  // ...
} 
```
looks simplier for me

File: packages/devextreme/js/ui/tab_panel.js [hunk 3]
```diff
@@ -252,6 +273,51 @@ const TabPanel = MultiView.inherit({
...
+
+        if([TABS_POSITION.right, TABS_POSITION.left].includes(tabsPosition)) {
+            return TABS_ORIENTATION.vertical;
+        }
+    },
+
+    _getTabPanelTabsPositionClass(value = null) {
+        value = value ?? this.option('tabsPosition');
+
+        switch(value) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/tab_panel.js]: Why we should have cases for it?
```
const TABPANEL_TABS_POSITION_CLASS = dx-tabpanel-tabs-position;

return `${TABPANEL_TABS_POSITION_CLASS}-${value}`
```

File: packages/devextreme/js/ui/tab_panel.js [hunk 4]
```diff
@@ -252,6 +273,51 @@ const TabPanel = MultiView.inherit({
...
+            default:
+                return TABPANEL_TABS_POSITION_CLASS.top;
+        }
+    },
+
+    _toggleTabPanelTabsPositionClass(prevPosition) {
+        const prevClass = this._getTabPanelTabsPositionClass(prevPosition);
+        const newClass = this._getTabPanelTabsPositionClass();
+
+        if(prevClass) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/tab_panel.js]: Whether it was or wasn't doesn't matter, we just need to clear the class

```
for (const position in TABS_POSITION) {
    this.$element().removeClass(this._getTabPanelTabsPositionClass(position));
}
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js [hunk 5]
```diff
@@ -228,6 +263,16 @@ QUnit.module('options', {
         assert.equal(this.tabWidgetInstance.option('selectedItem'), this.items[1], 'option <selectedItem> of nested tabs widget successfully changed');
     });
 
+    QUnit.test('orientation tabs option test', function(assert) {
+        assert.expect(2);
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js]: I don't think it makes any sense

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js [hunk 6]
```diff
@@ -228,6 +263,16 @@ QUnit.module('options', {
         assert.equal(this.tabWidgetInstance.option('selectedItem'), this.items[1], 'option <selectedItem> of nested tabs widget successfully changed');
     });
 
+    QUnit.test('orientation tabs option test', function(assert) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js]: Let's use more informative caption for test `orientation option should be passed to tabs correctly` or something like this

File: packages/devextreme/js/ui/tab_panel.js (hunk 1 above)
Comment (marker-dao) [reply packages/devextreme/js/ui/tab_panel.js:279]: We shouldn't use TABS_ORIENTATION here.

File: packages/devextreme/js/ui/tab_panel.js (hunk 2 above)
Comment (marker-dao) [reply packages/devextreme/js/ui/tab_panel.js]: My way is not difficult for understanding.

File: packages/devextreme/js/ui/tab_panel.js (hunk 3 above)
Comment (marker-dao) [reply packages/devextreme/js/ui/tab_panel.js]: There is not such class `dx-tabpanel-tabs-position`. We shouldn't use string templates for strict class variables. May be tomorrow we will use the different variable, so we will need to change js code.

File: packages/devextreme/js/ui/tab_panel.js (hunk 4 above)
Comment (marker-dao) [reply packages/devextreme/js/ui/tab_panel.js]: We shouldn't remove all TABS_POSITION classes in for cycle. Just a one class.

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js (hunk 5 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js]: I agree.

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js (hunk 6 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js]: Nice point, I will replace this caption.

File: packages/devextreme/js/ui/tab_panel.js (hunk 1 above)
Comment (marker-dao) [reply packages/devextreme/js/ui/tab_panel.js:279]: [feccc1664117de61365ba2df824d600fa6ed06f6]

File: packages/devextreme/js/ui/tab_panel.js (hunk 4 above)
Comment (marker-dao) [reply packages/devextreme/js/ui/tab_panel.js]: [feccc1664117de61365ba2df824d600fa6ed06f6]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js (hunk 6 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js]: [feccc1664117de61365ba2df824d600fa6ed06f6]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js (hunk 5 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js]: [feccc1664117de61365ba2df824d600fa6ed06f6]

**Review by EugeniyKiyashko — DISMISSED** [review]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js [hunk 7]
```diff
@@ -228,6 +263,14 @@ QUnit.module('options', {
         assert.equal(this.tabWidgetInstance.option('selectedItem'), this.items[1], 'option <selectedItem> of nested tabs widget successfully changed');
     });
 
+    QUnit.test('orientation option should be passed to tabs correctly', function(assert) {
+        assert.equal(this.tabWidgetInstance.option('orientation'), TABS_ORIENTATION.horizontal, 'option <orientation> successfully passed to nested tabs widget');
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js]: strictEqual

**Review by AlexanderMoiseev — DISMISSED** [review]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js (hunk 7 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/tabPanel.tests.js]: [776a6faeb29f6e3b14ddb5c72a437e7c41d046fe]

**Review by AlexanderMoiseev — DISMISSED** [review]

**Review by alexslavr — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by AlexanderMoiseev — APPROVED** [review]

**Review by vladaskorohodova — APPROVED** [review]

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**